src/types.ts

```diff
@@ -398,6 +404,12 @@ export enum PoolDirection {
   Reverse,
 }
 
+export enum HarvestType {
+  WithdrawAsSgene,
```

Comment:
1. Move `HarvestType` to `genepets.ts`

2. Rename:
- ~~`ClaimRewards`~~ => initialize
- ~~`Withdraw`~~ => completeAsGene
- ~~`WithdrawAsSgene`~~ => completeAsSGene